### PR TITLE
[WC-2001] Tree node on click issue

### PR DESCRIPTION
**Pull request by iobuhov**
### Pull request type

Bug fix (non-breaking change which fixes an issue)

New feature (non-breaking change which adds functionality)

---

### Description
A fix that address issue with on click in tree node.

### CODE REVIEW

File: packages/pluggableWidgets/tree-node-web/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 -   We fixed an issue with keyboard input for "form" widgets when they were put into a tree node. Up to this version, pressing "space" or "enter" in widgets like ‘Text box’ or ‘Text area’ was not handled properly, and keyboard input was ignored. The new version should have no issues with keyboard input in child widgets.
 
+-   We fixed an issue with the open/close state for the tree node. This issue was visible when any clickable widget was used to customize the tree node header. Starting from this version, you can choose which part of the header (icon or whole header), when clicked, triggers a collapsed or expanded state for the node.
+
+### Breaking changes
+
+-   We changed the widget HTML. As a result, some CSS selectors may be affected. Please check your custom stylus (if used) after updating to this version.
```
Comment (rahmanunver) [inline packages/pluggableWidgets/tree-node-web/CHANGELOG.md:17]: typo at 

> "...custom styles..."

File: packages/pluggableWidgets/tree-node-web/src/TreeNode.editorConfig.ts [hunk 2]
```diff
@@ -172,3 +179,17 @@ function getChevronIconPreview(headerType: HeaderTypeEnum, isDarkMode: boolean):
...
 }
+
+export function check(props: TreeNodePreviewProps): Problem[] {
+    const errors: Problem[] = [];
+
+    if (props.openNodeOn === "iconClick" && props.showIcon === "no") {
+        errors.push({
+            property: "openNodeOn",
+            message:
+                'The header icon is required to be visible when "Open node on" set to "Icon is clicked". Right now, the "Show icon" is set to "No".'
```
Comment (rahmanunver) [inline packages/pluggableWidgets/tree-node-web/src/TreeNode.editorConfig.ts:190]: tiny grammar fix, 

> "Open node on" **is** set to...

**Review by rahmanunver — APPROVED** [review]

File: packages/pluggableWidgets/tree-node-web/CHANGELOG.md (hunk 1 above)
Comment (iobuhov) [reply packages/pluggableWidgets/tree-node-web/CHANGELOG.md]: done

**Review by rahmanunver — APPROVED** [review]

**Review by samuelreichert — APPROVED** [review]
LGTM!

**Pull request merged**